keys/location.go

```diff
@@ -0,0 +1,29 @@
+package keys
+
+import "strings"
+
+type Location struct {
+	keyspace []string
+	joiner   string
+	name     string
+}
+
+func NewLocation(keyspace ...string, joiner, name string) *Location {
```

Comment:
I don't think there's anything wrong with allowing people to provide the key as a string directly. If they want to join that up with other strings, there are already functions in the standard library to do so.